### ISSUE TITLE
[css-overflow] Let 'overflow' accept two values

### ISSUE THREAD

**Issue by Loirooriol** [w3c/csswg-drafts#2484]
From https://github.com/w3c/csswg-drafts/issues/2000#issuecomment-346085897. `overflow` is a shorthand property of `overflow-x` and `overflow-y`, but it can only set them to the same value. It would be more convenient if the syntax was

    [ visible | hidden | clip | scroll | auto ]{1,2}

The two values specify the behavior in the horizontal and vertical direction, respectively. If only one value is specified, the second value defaults to the same value. For example,

```css
overflow: hidden auto; /* overflow-x: hidden; overflow-y: auto */
```

Then if #1282 introduces a flow-relative syntax for shorthands, it could apply to `overflow`, e.g.

```css
overflow: hidden auto !relative; /* overflow-inline: hidden; overflow-block: auto */
```

**Comment by frivoal** [comment]
I don't know how urgent this is, but it seems reasonable to me.

**Comment by fantasai** [comment]
Agreed with @frivoal’s comment, but there's the additional question of: should the ordering be physical or logical?

**Comment by CarterLi** [comment]
Just noticed this in [chromestatus](https://www.chromestatus.com/features/5090725653905408)

Can this new syntax allows `overflow-x: visible; overflow-y: hidden'?

eg. `overflow: visible hidden` won't be interpreted as `overflow: auto hidden`.

I think the old behavior was a design flow and made no sense. Now it's a chance to (partially) fix the mistake.

https://stackoverflow.com/questions/6421966/css-overflow-x-visible-and-overflow-y-hidden-causing-scrollbar-issue#answer-6433475

**Comment by emilio** [comment]
There's no change to the valid or invalid combinations of overflow-x/y.

**Comment by emilio** [comment]
The reason for it, is because the normal overflow-x / overflow-y adjustment doesn't (can't) happen at parse time, so there's no way to differentiate this syntax from the other at that stage unfortunately.

**Comment by fantasai** [comment]
I'm a bit concerned that this is not consistent with scroll-snap-align, which is also two-valued, and is much more closely related to overflow than background-position is.
